ndcube/utils/wcs.py

```diff
@@ -427,9 +427,13 @@ def array_indices_for_world_objects(wcs, axes=None):
     else:
         world_indices = np.arange(wcs.world_n_dim)
 
-    object_names = np.array([wao_comp[0] for wao_comp in wcs.world_axis_object_components])
+    try:
+        object_names = np.array([wao_comp[0] for wao_comp in wcs.world_axis_object_components])
+    except AttributeError:
+        object_names = np.array([wao_comp[0]
+                                 for wao_comp in wcs.low_level_wcs.world_axis_object_components])
 
-    array_indices = [[]] * len(set(object_names))
+    array_indices = [[]] * len(object_names)
```

Comment:
Pretty sure this needs to be a set still?

Reply:
Changing that was what fixed the bug.  The elements in which nothing was entered are removed by the tuple comprehension for the for loop.

Reply:
Can you add a comment above like 442 (atleast_1d) saying that it selects the first occurrence of the name.